lib/token.h

```diff
@@ -55,9 +55,9 @@ class Variable;
 class CPPCHECKLIB Token {
 private:
     Token **tokensBack;
+    Token **tokensFront;
```

Comment:
People are worried about the size of the token class. A single pointer is enough. We could have a struct:

    struct TokensFrontBack {
        Token **front;
        Token **back;
    };

And then the token class only needs a single pointer:

    TokensFrontBack  * const tokensFrontBack;

Reply:
The Token class already contains 20 pointer and int members.  Would one extra pointer make that much difference?

Reply:
Sorry.. I wanted that the TokenList would have a TokensFrontBack object. And then this would be passed to the Token constructor instead of _front and _back.

Reply:
> The Token class already contains 20 pointer and int members.

Yes it is "too much". People would like to have fewer so if we can cleanup in some clever way..

> Would one extra pointer make that much difference?

not much.

Reply:
Should TokenList have a TokensFrontBack member or a TokensFrontBack* member?

Reply:
I suggest that TokenList will have a `TokensFrontBack` member.